### ISSUE TITLE
Unable to get metrics in prometheus

### ISSUE THREAD

**Issue by aelmehra** [debianmaster/openshift-examples#52]
Hello,

Ive followed all of the instructions here but I am still not able to get any of the container metrics in prometheus. I am returning lots of other metrics such ass process_cpu_seconds_total, apiserver_audit_event_total, and kubelet_docker_operations to name a few. Is there something that needs to be done prior to this?

**Comment by debianmaster** [comment]
Not sure how to debug your issue.  but try seeing prometheus pod logs for any potential errors

**Comment by aelmehra** [comment]
No errors in the pods logs. Do I need to enable cluster metrics? [Enabling Cluster Metrics](https://docs.openshift.org/latest/install_config/cluster_metrics.html)

**Comment by toni-moreno** [comment]
I have cluster metrics enabled with heapster + hawkular

![image](https://user-images.githubusercontent.com/5883405/37784416-12188cc0-2df8-11e8-88dc-df30bd0006c7.png)

And I've created a new separated monitoring project  ( instead of deploying on the openshift-infra ).

![image](https://user-images.githubusercontent.com/5883405/37784642-92a13f54-2df8-11e8-9ac5-be25063dbd27.png)

No errors in any of the containers, and can see some data on the prometheus console

![image](https://user-images.githubusercontent.com/5883405/37789273-bf016d2a-2e03-11e8-8206-9df53b0a1d2b.png)


but nothing in the grafana dashboard.


![image](https://user-images.githubusercontent.com/5883405/37789308-d8e53794-2e03-11e8-94ca-c03b64961004.png)

Can you tell me what to do to fix?

OpenShift Master:
v3.7.1+282e43f-42
Kubernetes Master:
v1.7.6+a08f5eeb62

**Comment by toni-moreno** [comment]
I've recreated on openshift-infra project but does not work either.

I can see that there is some metrics on the Prometheus frontend but there is not any "container*" metric.

![image](https://user-images.githubusercontent.com/5883405/37846273-67fc2a72-2ecd-11e8-9b87-a830e0d32061.png)

**Comment by sleter** [comment]
I have exact same problem. I can't see all the metrics in Prometheus.

Openshift 3.9

**Comment by debianmaster** [comment]
@toni-moreno  @sleter sorry i missed your message.   how did you install prometheus on openshift?

better way is to use .  ansible playbooks   rather than configuring manually.       like this 

```sh
ansible-playbook openshift-ansible/playbooks/openshift-prometheus/config.yml
```

**Comment by debianmaster** [comment]
its pretty straight forward no tweaks needed.

**Comment by sleter** [comment]
@debianmaster Thanks for reply! I will give it a go

**Comment by aelmehra** [comment]
It looks like openshift-ansible/playbooks/openshift-prometheus/config.yml deploys the prometheus rules with most of the container_ metrics off

This is a excerpt from the configmap prometheus.yml


```
  # Exclude a set of high cardinality metrics that can contribute to significant
  # memory use in large clusters. These can be selectively enabled as necessary
  # for medium or small clusters.
  metric_relabel_configs:
  - source_labels: [__name__]
    action: drop
    regex: 'container_(cpu_user_seconds_total|cpu_cfs_periods_total|memory_usage_bytes|memory_swap|memory_working_set_bytes|memory_cache|last_seen|fs_(read_seconds_total|write_seconds_total|sector_(.*)|io_(.*)|reads_merged_total|writes_merged_total)|tasks_state|memory_failcnt|memory_failures_total|spec_memory_swap_limit_bytes|fs_(.*)_bytes_total|spec_(.*))'
```

After I changed the action from drop to keep all of the metrics showed up.

**Comment by sleter** [comment]
Thanks guys! I installed openshift-metrics and openshift-prometheus with ansible and now I can see all the metrics.

**Comment by sleter** [comment]
Closes #52
